### ISSUE TITLE
Save the specified reactor in ReactorSelectionMixin for later inspection

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4563]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4563|
|Type|enhancement|
|Created|2010-07-09 20:33:25Z|
|Branch|https://github.com/twisted/twisted/tree/save-reactor-option-4563|

`ReactorSelectionMixin` gives an `Options` subclass options for listing and installing a particular reactor.  To ensure that the reactor has the greatest chance of being successfully installed, it installs it while the option parsing is happening.  This is cool.

However, it means that no "reactor" key is ever added to the `Options` instance, so it's hard to tell which reactor was installed.

In particular, `disttrial` ([#1784](https://github.com/twisted/twisted/issues/1784)) is interested in this so that it can configure the test runner processes it creates to use the same reactor.

<details><summary>Searchable metadata</summary>

```
trac-id__4563 4563
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_save_reactor_option_4563 branches/save-reactor-option-4563
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1278707605000000 1278707605000000
changetime__1279056429000000 1279056429000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29745]) Branching to 'save-reactor-option-4563'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Pulled the change out of the disttrial branch and added a test and documentation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [29750]) Drop this, see [#4563](https://github.com/twisted/twisted/issues/4563)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29818]) Merge save-reactor-option-4563

Author: exarkun
Reviewer: therve
Fixes: [#4563](https://github.com/twisted/twisted/issues/4563)

Enhance twisted.application.app.ReactorSelectionMixin so that it saves
the selected reactor name for later inspection.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
